urllib3/util/connection.py

```diff
@@ -99,3 +106,39 @@ def _set_socket_options(sock, options):
 
     for opt in options:
         sock.setsockopt(*opt)
+
+
+def family_filter():
```

Comment:
`allowed_gai_family()`? Conveys the context (`gai`) and makes clear that it's restricting some options.

:+1: on the change to AF_UNSPEC; didn't come across that constant when initially setting this up.